addons/account/models/account_move.py

```diff
@@ -703,6 +703,8 @@ def _compute_date(self):
                 self.env.add_to_compute(move.line_ids._fields['date'], move.line_ids)
                 # might be protected because `_get_accounting_date` requires the `name`
                 self.env.add_to_compute(self._fields['name'], move)
+            # Highest name should be changed as the sequence date (date field) changes
+            move._compute_highest_name()
```

Comment:
FYI `highest_name` is a non-stored computed field. So the compute is called everytime the field is accessed. Calling the compute explicitly from the code will not have any effect.